mailpoet/lib/Util/APIPermissionHelper.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace MailPoet\Util;
+
+// \WP_REST_Posts_Controller is always available in WP runtime but not when building the plugin outside of WP env
+if (class_exists("\WP_REST_Posts_Controller")) {
```

Comment:
> Could we maybe do this (see below) instead of the conditional class definition?
> 
> ```
> require_once ABSPATH . '/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php';
> ```
> 
> We [already have a couple of these](https://github.com/mailpoet/mailpoet/search?q=require_once+ABSPATH). I'm curious how DI dump actually works when the class is only conditional 🤔

I actually tried the `require_once` approach first, but PHPStan was not happy with it, this if statement approach works and I've verified the `FreeCachedContainer ` has `APIPermissionHelper ` in the build artifact, but still would be happy to learn if you think this approach is error prone